release_docs/RELEASE.txt

```diff
@@ -731,20 +731,6 @@ Support for new platforms, languages and compilers
     
 Bug Fixes since HDF5-1.14.0 release
 ===================================
-    Configuration:
-    -------------
-    - Fix Autotools -Werror cleanup
```

Comment:
Moved this entry down to the existing "Configuration" section and removed the duplicate header added above.